### PR TITLE
Spelling fixes in raster addons documentation

### PR DESCRIPTION
**Pull request by echoix**
Following some work already pulled recently, I continued proofreading the documentation and strings from the documentation, staying in the raster modules. Since I've stopped working on it without finishing the 167 subfolders (41 folders done on 167), I think it is still valuable to fine-tune and pull these changes, since I still passed through 20k lines of code.

All modules from `r.accumulate` to `r.futures` (alphabetically), inclusively, were proofread and changed if needed, except `r.estimap.recreation` and `r.damflood`. I skipped `r.estimap.recreation` since it appears to be developed on an external GitLab repository, so fetching upstream changes are more appropriate. For `r.damflood`, the changes are bigger and more complex, they are in another branch and the pull request will follow.

I saw that the `r.edm.eval` add-on is only a shell script, that seems to be what add-ons were in GRASS 6 if I understood correctly. Does that mean that it is incompatible with GRASS 7 or GRASS 8?

I have some other notes that I took on a paper when working on this, but unfortunately I don't have the paper with me at the moment, so I will add the revision comments to the specific lines where I wasn't sure either tomorrow or later, anyway this PR should take a couple days to review.

### CODE REVIEW

**Comment by ninsbl** [comment]
Thanks @echoix  for your great effort!

I actually got inspired by your work and checked what options there are to spellcheck my contributions (as i tend to overlook typos). So, I came across a tool names [typos](https://github.com/crate-ci/typos), that would even allow to spellcheck code (e.g. variable names) and could be run in github actions.

As the GRASS codebase has quite some history, some typos have accumulated (as you noticed). The `typos` tool has no interactive mode but could be applied as described here:
https://github.com/crate-ci/typos/issues/397
in a first run.

Just in case this could be of interest...

**Comment by echoix** [comment]
Ah, that could be an idea to test as CI. For now I was using Antidote, a tool recognized for its French correcting tool for 25+ years, but also has English support since 7 years (both languages in a same document). I use this tool commonly for years for all school assignments etc, but it's not adapted to code, except Tex.

But I take great pleasure reviewing this, since the algorithms are rather small and easy to understand, and I can learn a lot just by reading it. And at the same time, I have the opportunity to view what to fix to be consistant throughout the modules, which never happens when each one is written by someone different. (Like the formatting of the tags, someone, either you or Markus reminded me and sent a link to a guide)

When reading some of the pages, I had an idea for an improvement. Since the SVN date isn't used, and commented by default (should I revive it completely when passing?), the "updated" date could be generated by the page builder. However I don't know exactly what date should be the most appropriate: the commit date can be changed/faked, and the build date is irrelevant if the same document is built even without changes (or if the global style changes, it isn't a content update). It's something to think about...

**Comment by wenzeslaus** [comment]
+1 for the spell checks test in the CI. For example, MegaLinter has cspell and misspell.

**Comment by echoix** [comment]
> +1 for the spell checks test in the CI. For example, MegaLinter has cspell and misspell.

Indeed, I checked it up, MegaLinter seems a good upgrade to the project. It seems that MegaLinter was based off a rejected PR of Super-Linter, which is the GitHub general linter used here. So, that would mean replacing this, plus the Black and flake8 actions, since they are already included?

I'm thinking of using it locally for the following checks, if possible. However, many `.c` files were not formatted consistently at all, so it might need a separate PR. I remember fighting my VScode to not include the "fixed" indentation when posting back the fixes, since some mixed tab and spaces (in the same line) were used. I wanted to keep the diffs clean.

**Comment by ecodiv** [comment]
Great work! I quickly went through some of the commits, but I am not sure how the review process works. One can review and comment or approve each commit separately? If I open a commit, there is this green button 'Review changes'. Is that button about the commit I opened, or the whole list?

**Comment by ecodiv** [comment]
> I saw that the `r.edm.eval` add-on is only a shell script, that seems to be what add-ons were in GRASS 6 if I understood correctly. Does that mean that it is incompatible with GRASS 7 or GRASS 8?
> 
That is one I still wanted to port to Python, but to be honest sort of forgot about. It is mostly R code, should not be too difficult to change to Python. I'll have a look at it.

**Comment by neteler** [comment]
> One can review and comment or approve each commit separately?

Yes. You can comment on each line or propose edits:

image

**Comment by neteler** [comment]
> Following some work already pulled recently, I continued proofreading the documentation and strings from the documentation

Thanks for your great work.

BTW: in GRASS-core, we also have the script `codespell.py` borrowed from GDAL, see https://github.com/OSGeo/grass/blob/main/utils/fix_typos.sh (I used it once, time ago).

**Comment by neteler** [comment]
> However, many `.c` files were not formatted consistently at all, so it might need a separate PR.

For this, we have
- https://github.com/OSGeo/grass/blob/main/utils/grass_indent.sh
- https://github.com/OSGeo/grass/blob/main/utils/grass_indent_ALL.sh

They are naturally pretty invasive, so some care needs to be taken with the open large pull requests (maybe skip indentation of those for now till they have been merged).

**Comment by echoix** [comment]
> They are naturally pretty invasive, so some care needs to be taken with the open large pull requests (maybe skip indentation of those for now till they have been merged).

Of course. I was thinking of maybe having a preparatory pull request if we were to have a CI that would make sure it doesn't get dirty again.

**Comment by echoix** [comment]
Ok, for my notes that I took, finally there isn't much more than the sentence in r.edm.eval, and the modules that I skipped.
The remaining fix was a spelling fix in the r.extract.py append_random() that should be fixed upstream since it was copied from the core repo, and a reminder to enter an issue to flag that r.diversity might not work in grass 8 since the code explicitely makes extra checks and builds a path with "`GRASS7`" in its `main()`

File: src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html
Comment (echoix) [inline src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html:88]: I have a style question for @neteler, concerning my understanding of how these dates were abandoned in the link you have pointed me on another PR https://trac.osgeo.org/grass/wiki/Submitting/Docs#Footer .
Since it is not useful anymore, and when I change the file the existing "Last changed" date is now invalid, it is the good way to go to comment it, or should I remove them of the code? Commented out is pointless, except for maybe git blame that is cleaner to find? Is the commented HTML stripped out of the generated web page? If not, it might be useful to remove when applicable to prevent any potential parsing oddities, or for search engines if ever they index this.

**Comment by echoix** [comment]
> Great work! I quickly went through some of the commits, but I am not sure how the review process works. One can review and comment or approve each commit separately? If I open a commit, there is this green button 'Review changes'. Is that button about the commit I opened, or the whole list?

You can open the commits or the files, and press the "+" like @neteler showed. A comment or suggested change can be made for the cumulative code changed in the PR (like if additionnal commits change something and it is still not right). See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request if you want to learn more. :)

File: src/raster/r.fill.category/r.fill.category.html
Comment (echoix) [inline src/raster/r.fill.category/r.fill.category.html:79]: Forgot to ignore this unnecessary diff from the commit.
```suggestion

```

File: src/raster/r.findtheriver/r.findtheriver.html
Comment (echoix) [inline src/raster/r.findtheriver/r.findtheriver.html:50]: Forgot to ignore this unnecessary diff from the commit.
```suggestion

```

File: src/raster/r.flexure/r.flexure.py
Comment (echoix) [inline src/raster/r.flexure/r.flexure.py:113]: Forgot to ignore this unnecessary diff from the commit.
```suggestion

```

File: src/raster/r.flexure/r.flexure.html
Comment (echoix) [inline src/raster/r.flexure/r.flexure.html:66]: Forgot to ignore this unnecessary diff from the commit.
```suggestion

```

**Review by echoix — COMMENTED** [review]
I found unnecessary lines changed in a diff.

File: src/raster/r.area.createweight/r.area.createweight.py
Comment (echoix) [inline src/raster/r.area.createweight/r.area.createweight.py:174]: I went back in the scikit learn docs to correct this, but was this unique abbreviation of the concept made to adapt to GUI size constraints?

File: src/raster/r.droka/r.droka.py
Comment (echoix) [inline src/raster/r.droka/r.droka.py:223]: Can an Italian-speaking contributor approve this translation? I couldn't understand the original sentence using translation tools and the context.

Comment (echoix) [inline src/raster/r.droka/r.droka.py:223]: Can an Italian-speaking contributor approve this translation? I couldn't understand the original sentence using translation tools and the context.

File: src/raster/r.droka/r.droka.py
Comment (echoix) [inline src/raster/r.droka/r.droka.py:243]: I couldn't find a translation that I was confident with.

File: src/raster/r.edm.eval/r.edm.eval.html
Comment (echoix) [reply src/raster/r.edm.eval/r.edm.eval.html]: Not outdated

**Comment by neteler** [comment]
> a reminder to enter an issue to flag that `r.diversity` might not work in grass 8 since the code explicitly makes extra checks and builds a path with "`GRASS7`" in its `main()`.

(hopefully addressed in #667)

File: src/raster/r.edm.eval/r.edm.eval.html
Comment (neteler) [reply src/raster/r.edm.eval/r.edm.eval.html]: @ecodiv: please advise here.

File: src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html
Comment (neteler) [reply src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html:88]: Well, we still hope to invent a way to reactivate these file dates with git. That's why they are still there.
Suggestions are welcome.

In any case, there must be no date but just the variable. I have fixed above line in #667

File: src/raster/r.edm.eval/r.edm.eval.html
Comment (ecodiv) [reply src/raster/r.edm.eval/r.edm.eval.html]: The sentence isn't terribly clear, but correct in essence.  I hope to find time to look at this add-on soon, and will probably so some rewriting of the text. For now, I would leave it as it is.

File: src/raster/r.droka/r.droka.py
Comment (veroandreo) [reply src/raster/r.droka/r.droka.py:223]: My interpretation is that it uses the points in the raster for start_coordinates in the following r.cost command. So, I'd translate as: ```Cost calculation (split and use the starting points in start_raster at the start_coordinates parameter in r.cost call)```. Does that make sense? Note that I removed the extra parenthesis introduced.

Comment (echoix) [reply src/raster/r.droka/r.droka.py:223]: The extra parentheses were because I left the original Italian comments there and added my suggestion underneath (I don't speak Italian)

Comment (veroandreo) [reply src/raster/r.droka/r.droka.py:223]: Ah, didn't realize you planned to leave the original comment too

Comment (echoix) [reply src/raster/r.droka/r.droka.py:223]: No, you're right, I wanted to replace them :), In fact, I wanted to replace to English for all the file, but I didn't want to replace it with something that lost a meaning

File: src/raster/r.droka/r.droka.py
Comment (echoix) [reply src/raster/r.droka/r.droka.py:243]: The `$ang` seems to be an input parameter, which is called shadow angle in the descriptions, https://grass.osgeo.org/grass80/manuals/addons/r.droka.html

> DESCRIPTION
> The intersection between the topographic profile and the line starting from the point of detachment with a certain angle to the horizontal line defines the point of maximum propagation (Evans and Hungr, 1993; Jaboyedoff and Labiouse, 2003). **The angle of inclination and the line of propagation are defined as zenith angle and shadow zone.** This script defines rockfall zones from a digital elevation model (DEM) and vector layer containing starting point or points.


> Shadow angle = integer
> **Angle of inclination** that defines the propagation zone. It's calculated from an horizontal plane starting from source point: highest values determine a lowest propagation.


If ever you can understand the explications around page 40/155 of this document (it is the same formula with text around), it might help to confirm. https://va.minambiente.it/File/Documento/98213

File: src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html
Comment (echoix) [reply src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html:88]: Can I leave the commit changes as-it for this PR, or I need to revert to avoid merge conflicts? One of the two PRs will need to handle a merge conflict anyways.

File: src/raster/r.area.createweight/r.area.createweight.py
Comment (echoix) [reply src/raster/r.area.createweight/r.area.createweight.py:174]: I assume there are no problems then.

**Comment by echoix** [comment]
After resolving the last two opened conversations, I think the PR can be merged.

File: src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html
Comment (neteler) [reply src/raster/r.futures/r.futures.devpressure/r.futures.devpressure.html:88]: Since this is larger, I'll take care of the merge conflicts in my smaller PR.

File: src/raster/r.area.createweight/r.area.createweight.py
Comment (neteler) [reply src/raster/r.area.createweight/r.area.createweight.py:174]: Your change is fine.

**Review by neteler — APPROVED** [review]

**Comment by wenzeslaus** [comment]
Sorry, unrelated to this already long PR discussion, but anyway:

> > +1 for the spell checks test in the CI. For example, MegaLinter has cspell and misspell.
> 
> So, that would mean replacing [Super-Linter]...

Maybe, maybe not, my guess is that MegaLinter and Super-Linter diverged now enough that each has its own merits.

> ...plus the Black and flake8 actions, since they are already included?

Not necessarily, direct and possibly finer control over Flake8 and Black (and possibly Pylint) is likely desired, e.g., due to changes in Black versions.

**Comment by neteler** [comment]
If there are no objections, I'll merge this PR. Thanks for your efforts @echoix!

**Comment by echoix** [comment]
Looks good to me! I will be ready to continue next weekend!

**Comment by echoix** [comment]
> +1 for the spell checks test in the CI. For example, MegaLinter has cspell and misspell.

Jusr to let you know that its been two weekends I'm playing around with MegaLinter in my fork. I'm learning how it works and configuring it correctly to be confident about it when it will be time to make a PR. Last week, the maintainer of MegaLinter jumped into my test PR, and made some performance changes to MegaLinter itself with what I observed. He even forked my fork to test an even more optimal configuration. I'm writing it here just to not make some duplicated work if someone though of implementing it while I'm working on it. My goal is to try it out, have it here, and one day have it in the core repo too.

**Pull request merged**
